src/routes/ProtectedRoute.tsx

```diff
@@ -1,11 +1,30 @@
+import { ProtectedRouteProps } from '@/util/interfaces'
 import { Navigate, Outlet } from 'react-router-dom'
 
-export function ProtectedRoute() {
+export function ProtectedRoute({ allowedRoles }: ProtectedRouteProps) {
     const isAuthenticated = !!localStorage.getItem('token')
 
+    let userRole = null
+    try {
+        const user = localStorage.getItem('user')
+        if (user) {
+            const userData = JSON.parse(user)
+            userRole = userData.roles
+        }
+    } catch (error) {
+        console.error('Failed to parse user data:', error)
+        localStorage.removeItem('user')
+        localStorage.removeItem('token')
+        return <Navigate to="/login" replace />
+    }
```

Comment:
I would recommend putting the above the logic inside a resuable function.

eg:

```ts
function getAuthInfo() {
  const token = localStorage.getItem('token');
  const isAuthenticated = !!token;

  if (!isAuthenticated) {
    return { isAuthenticated: false, userRole: null };
  }

  let userRole = null;

  try {
    const user = localStorage.getItem('user');
    if (user) {
      const userData = JSON.parse(user);
      userRole = userData.roles;
    }
  } catch (error) {
    console.error('Failed to parse user data:', error);
    localStorage.removeItem('user');
    localStorage.removeItem('token');
  }

  return { isAuthenticated, userRole };
}
```


Usage example

eg: 
```ts
const { isAuthenticated, userRole } = getAuthInfo();
```